### DISCUSSION TITLE
Open Graph Image link for Production Deployment

### DISCUSSION THREAD

**Discussion by 3ru** [vercel/community#7199] [Feature Question] [answered]
### Question

I'm not sure if this is a bug or expected behavior, so I'm asking for clarification.

When `Vercel Authentication` is enabled for Deployment Protection in a private repository, the Deployment Link is protected. As a result, only the Production Deployment is made public.
This is ideal up to this point.

However, the `og:image` link on the Production Deployment site uses the **Deployment Link**. As a result, although the site itself is properly published, the og image is not displayed.
![image](https://github.com/vercel/community/assets/69892552/70300bb9-9e68-442f-b11c-4022dba9691b)

![image](https://github.com/vercel/community/assets/69892552/a293b59c-6784-41bc-aa43-ab2117019b36)

Is this the expected behavior?
The workaround is to turn off Vercel Authentication.

The particularly problematic issue is that when checking the "Open Graph" tab on the Build Logs screen on Vercel, it appears as if the OGP is being displayed correctly.

![image](https://github.com/vercel/community/assets/69892552/431bc1c1-a064-4fdd-941f-f8fdc358edee)

**Comment by avemeva** [comment]
I have the same issue, have u been able to figure it out? @3ru

**Reply by 3ru** [reply]
@andrewshvv
Not yet😿

**Reply by avemeva** [reply]
Might be related
https://usenextbase.com/docs/nextjs/v2/developer-faq/why-next-public-site-url

**Reply by avemeva** [reply, marked as answer]
Yes, setting `NEXT_PUBLIC_SITE_URL` in `vercel` seems to have done the trick.

```
export async function generateMetadata({
  params,
}: {
  params: { slug: string };
}): Promise<Metadata> {
  ....

  return {
    ...
    metadataBase: new URL(getBaseUrl()),
    ...
    },
  };
}
```

```
export const getBaseUrl = () => {
  if (process.env.NEXT_PUBLIC_SITE_URL) {
    return process.env.NEXT_PUBLIC_SITE_URL;
  }

  if (process.env.VERCEL_URL) {
    return `https://${process.env.VERCEL_URL}`;
  }

  return 'http://localhost:3000';
};
```

[a comment by an automated account is not shown]
